.buildkite/x-pack/pipeline.xpack.otel.yml

```diff
@@ -0,0 +1,103 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+name: "beats-xpack-otel"
+
+env:
+  AWS_ARM_INSTANCE_TYPE: "m6g.xlarge"
+  AWS_IMAGE_UBUNTU_ARM_64: "platform-ingest-beats-ubuntu-2204-aarch64"
+
+  IMAGE_UBUNTU_X86_64: "family/platform-ingest-beats-ubuntu-2204"
+
+  IMAGE_BEATS_WITH_HOOKS_LATEST: "docker.elastic.co/ci-agent-images/platform-ingest/buildkite-agent-beats-ci-with-hooks:latest"
+
+  GCP_DEFAULT_MACHINE_TYPE: "c2d-highcpu-8"
+
+  #Deps
+  ASDF_MAGE_VERSION: 1.15.0
+
+  # Unit tests
+  RACE_DETECTOR: "true"
+  TEST_COVERAGE: "true"
+
+steps:
+  - group: "check"
+    key: "x-pack-otel-check"
+
+    steps:
+      - label: "x-pack/otel: Run check and pre-commit"
+        command: {{matrix}}
+        retry:
+          automatic:
+            - limit: 1
+        agents:
+          image: "${IMAGE_BEATS_WITH_HOOKS_LATEST}"
+          cpu: "4000m"
+          memory: "8Gi"
+          useCustomGlobalHooks: true
+        notify:
+          - github_commit_status:
+              context: "x-pack/otel: run check and pre-commit"
```

Comment:
IIUC, this will create twice the same GH check, is that the expected behaviour? or one check per command here?

Otherwise, maybe the group could be used for notifying the status for both steps?